src/main/groovy/io/openliberty/tools/gradle/tasks/GenerateFeaturesTask.groovy

```diff
@@ -335,22 +335,20 @@ class GenerateFeaturesTask extends AbstractFeatureTask {
         project.configurations.compileClasspath.allDependencies.each {
             dependency ->
                 if (dependency.group.equals("javax") && dependency.name.equals("javaee-api")) {
-                    if (dependency.version.startsWith("8.")) {
-                        eeVersion = BINARY_SCANNER_EEV8
-                    } else if (dependency.version.startsWith("7.") && (isLatestVersion(eeVersion, BINARY_SCANNER_EEV7))) {
-                        eeVersion = BINARY_SCANNER_EEV7
-                    } else if (dependency.version.startsWith("6.") && (isLatestVersion(eeVersion, BINARY_SCANNER_EEV6))) {
-                        eeVersion = BINARY_SCANNER_EEV6
+                    String newVersion = composeEEVersion(dependency.version)
+                    if (newVersion != null && isLatestVersion(eeVersion, newVersion)) {
+                        eeVersion = newVersion
                     }
                 } else if (dependency.group.equals("jakarta.platform") &&
```

Comment:
I was traumatised some years ago when I wrote a conditional over ten lines long so I try to avoid long ones but since we are doing two compares and two more compares I'll go with you this time.